hist/hist/src/TH1Merger.cxx

```diff
@@ -383,16 +383,18 @@ Bool_t TH1Merger::BufferMerge() {
                fH0->Fill(hist->fBuffer[2*i + 2], hist->fBuffer[2*i + 1]);
          }
          if (fH0->fDimension == 2) {
-            auto h2 = dynamic_cast<TH2*>(fH0);
-            R__ASSERT(h2); 
-            for (Int_t i = 0; i < nbentries; i++)
-               h2->Fill(hist->fBuffer[3*i + 2], hist->fBuffer[3*i + 3],hist->fBuffer[3*i + 1] );
+            if (auto h2 = dynamic_cast<TH2*>(fH0)){
+                  R__ASSERT(h2); 
```

Comment:
Coverity has no way of knwing that `::Fatal()` never returns. We can tell it about `::Fatal()`, though. See https://coverity.cern.ch/docs/en/cov_checker_ref.html#killpath

I'll take care of it...